b2sdk/bucket.py

```diff
@@ -546,11 +547,30 @@ def upload_local_file(
         :param b2sdk.v2.EncryptionSetting encryption: encryption settings (``None`` if unknown)
         :param b2sdk.v2.FileRetentionSetting file_retention: file retention setting
         :param bool legal_hold: legal hold setting
+        :param b2sdk.v2.UploadMode upload_mode: desired upload mode
         :rtype: b2sdk.v2.FileVersion
         """
         upload_source = UploadSourceLocalFile(local_path=local_file, content_sha1=sha1_sum)
-        return self.upload(
-            upload_source,
+        sources = [upload_source]
+        large_file_sha1 = sha1_sum
+
+        if upload_mode == UploadMode.INCREMENTAL:
+            try:
+                existing_file_info = self.get_file_info_by_name(file_name)
+            except FileNotPresent:
+                pass
+            else:
```

Comment:
This can be rewritten in a cleaner manner with use of `suppress` and without use of elusive `try-except-else`.